src/main/java/com/google/api/codegen/transformer/SchemaTypeFormatter.java

```diff
@@ -26,12 +26,16 @@
   /** Get the full name for the given short name, using the default package. */
   String getImplicitPackageFullNameFor(String shortName);
 
-  /** Get the full name for the given type. */
-  String getFullNameFor(Schema type);
+  /** Returns the full name for the given type (without adding the full name to the import set). */
+  String getFullNameFor(String escapedSchemaName, Schema type, String parentName);
```

Comment:
I want to understand `escapedSchemaName` better - let's discuss in person tomorrow. Also I have an idea for a different way to handle imports in toolkit.